### PR TITLE
Add an HDF5 HL extern interface package

### PR DESCRIPTION
**Pull request by daviditen**
This is the output of a slightly modified c2chapel running on hdf5_hl.h from
HDF5 1.10.1_2.  The modifications to c2chapel were to better handle enums and
anonymous struct variables.

I then added functions to mimic a bunch of #defined Macros from hdf5_hl.h and
files it includes to the sub-module HDF5_HL_Macros.

I made a few edits where there were some mis-translations by c2chapel such as
not defining and using a type for some enums, or missing the translation from
"signed long long" to "c_longlong".

I also added tests that are translations of the examples from the HDF5 LITE
section of a Learning HDF5 tutorial.  These tests are skipped if libhdf5 can't
be located on the system.

### CODE REVIEW

File: modules/packages/HDF5_HL.chpl [hunk 1]
```diff
@@ -0,0 +1,3299 @@
...
+
+  extern proc H5Rget_name(loc_id : hid_t, ref_type : H5R_type_t, ref_arg : c_void_ptr, name : c_string, size : size_t) : ssize_t;
+
+  extern proc H5Rget_obj_type1(id : hid_t, ref_type : H5R_type_t, _ref : c_void_ptr) : H5G_obj_t;
+
+  extern proc H5Rdereference1(obj_id : hid_t, ref_type : H5R_type_t, ref_arg : c_void_ptr) : hid_t;
+
+  extern proc H5Screate(type_arg : H5S_class_t) : hid_t;
+
+  extern proc
```
[the remark this answers is not in the document]
Comment (daviditen) [reply modules/packages/HDF5_HL.chpl]: OK.

[the remark this answers is not in the document]
Comment (vasslitvinov) [reply modules/packages/HDF5_HL.chpl]: We had a way to declare the C name for a Chapel proc, something like `extern "foo" proc bar() ...`

Could we apply that same trick there?

This PR can go in without that, though.

[the remark this answers is not in the document]
Comment (daviditen) [reply modules/packages/HDF5_HL.chpl]: That trick currently only works for a externs at the top-level.  This is a field declared inside of an extern record, which doesn't have an `extern` keyword itself.  So there isn't a way to rename it that I know of.

**Comment by daviditen** [comment]
> I imagine Chapel users wanting to use Chapel strings instead of C strings as actual arguments when calling into HDF5. Is this something we should discuss? create an issue for it if appropriate?

The plan is to wrap this in a higher-level Chapel library that will take care of that and provide more high level features.

> If appropriate, please add .cleanfiles or CLEANFILES and additional entries to .gitignore for the .h5 files that the tests use.

Done.

**Comment by daviditen** [comment]
Thanks for reviewing @vasslitvinov!

**Pull request merged**